modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/sections/shared-components/BaseAction.js

```diff
@@ -0,0 +1,247 @@
+					type="text"
+					value={name}
+				/>
+			</ClayForm.Group>
+			<ClayForm.Group>
+				<label htmlFor="description">
+					{Liferay.Language.get('description')}
+				</label>
+
+				<ClayInput
+					id="description"
+					onBlur={() =>
+						updateActionInfo({
+							description,
+							executionType,
+							name,
+							priority,
+							template,
+						})
+					}
+					onChange={({target}) => {
+						setDescription(target.value);
+					}}
+					type="text"
+					value={description}
+				/>
+			</ClayForm.Group>
+			<ClayForm.Group>
+				<label htmlFor="template">
+					{`${Liferay.Language.get(
+						'template'
+					)} (${Liferay.Language.get('groovy')})`}
+
+					<span className="ml-1 mr-1 text-warning">*</span>
+				</label>
+
+				<ClayInput
+					component="textarea"
+					id="template"
+					onBlur={() =>
+						updateActionInfo({
+							description,
+							executionType,
+							name,
+							priority,
+							template,
+						})
+					}
+					onChange={({target}) => {
+						setTemplate(target.value);
+					}}
+					placeholder={placeholderTemplate}
+					type="text"
+					value={template}
+				/>
+			</ClayForm.Group>
+			<ClayForm.Group>
+				<label htmlFor="execution-type">
+					{Liferay.Language.get('execution-type')}
+				</label>
+
+				<ClaySelect
+					aria-label="Select"
+					id="execution-type"
+					onChange={({target}) => {
+						setExecutionType(target.value);
+					}}
+					onClickCapture={() =>
+						updateActionInfo({
+							description,
+							executionType,
+							name,
+							priority,
+							template,
+						})
+					}
+				>
+					{executionTypeOptions.map((item) => (
+						<ClaySelect.Option
+							key={item.value}
+							label={item.label}
+							selected={item.value === executionType}
+							value={item.value}
+						/>
+					))}
+				</ClaySelect>
+			</ClayForm.Group>
+			<ClayForm.Group>
+				<label htmlFor="priority">
+					{Liferay.Language.get('priority')}
+				</label>
+
+				<span
+					className="ml-1"
+					title={Liferay.Language.get('label-name')}
+				>
+					<ClayIcon
+						className="text-muted"
+						symbol="question-circle-full"
+					/>
+				</span>
+
+				<ClayInput
+					aria-label="Select"
+					id="priority"
+					min={MIN_PRIORITY}
+					onBlur={({target}) => {
+						let {value: newValue} = target;
+
+						newValue = limitValue({
+							defaultValue: DEFAULT_LIMIT,
+							min: MIN_PRIORITY,
+							value: newValue,
+						});
+
+						setPriority(newValue);
+
+						updateActionInfo({
+							description,
+							executionType,
+							name,
+							priority,
+							template,
+						});
+					}}
+					onChange={({target}) => {
+						let {value: newValue} = target;
+						newValue = newValue.includes('-')
+							? newValue.replace('-', '')
+							: newValue;
+
+						setPriority(newValue);
+					}}
+					type="number"
+					value={priority}
+				/>
+			</ClayForm.Group>
+			<div className="sheet-subtitle" />
+		</>
+	);
+};
+
+BaseAction.propTypes = {
```

Comment:
@eliasantosf Please, update the PropTypes of this file and also of the other files like TimerAction, TimerFields, TimerDuration, TimerInfo etc.